File: Python/Signaux/7.1.py
import os
import signal
import time

pid = os.fork()

if pid == 0:
    # Fils : calcul infini
    x = 0
    while True:
        x += 1
else:
    # Père : menu
    while True:
        choix = input("Menu : s=stop, r=resume, q=quit\n")
        if choix == 's':
            os.kill(pid, signal.SIGSTOP)
        elif choix == 'r':
            os.kill(pid, signal.SIGCONT)
        elif choix == 'q':
            os.kill(pid, signal.SIGKILL)
            break

File: C/Signaux/7.1.c
#include <stdio.h>
#include <unistd.h>
#include <stdlib.h>
#include <signal.h>

int main() {
    pid_t pid = fork();

    if(pid == 0) {
        // Fils : calcul sans fin
        int x = 0;
        while(1) {
            x += 1; // calcul simple
        }
    } else {
        // Père : menu de contrôle
        char choix;
        while(1) 
        {
            printf("Menu : s = stop, r = resume, q = quit\n");
            scanf(" %c", &choix); // espace avant %c pour consommer \n
            if(choix == 's') {
                kill(pid, SIGSTOP); // endort le fils
            } 
            else if(choix == 'r') {
                kill(pid, SIGCONT); // réveille le fils
            } 
            else if(choix == 'q') {
                kill(pid, SIGKILL); // tue le fils
                break;
            }
        }
    }

    return 0;
}